datafusion/core/src/datasource/listing_table_factory.rs

```diff
@@ -73,6 +73,8 @@ impl TableProviderFactory for ListingTableFactory {
                 CsvFormat::default()
                     .with_has_header(cmd.has_header)
                     .with_delimiter(cmd.delimiter as u8)
+                    .with_quote(cmd.options.get("quote").map_or(b'"', |x| x.as_bytes()[0]))
+                    .with_escape(cmd.options.get("escape").map_or(None, |x| Some(x.as_bytes()[0])))
                     .with_file_compression_type(file_compression_type),
```

Comment:
I'm not sure if this is the best way, maybe we can open another issue to improve it

Reply:
TBH I'd prefer to file a separate issued and integrate `QUOTE` & `ESCAPE` into `CreateExternalTable` pipeline. My concern here is that it probably will be confusing for end-users to pass read options into clause reserved for write options (and without any effect on writing CSV).